lib/widgets/downloadable_resources_page.dart

```diff
@@ -0,0 +1,293 @@
+  State<DownloadableResourcesPage> createState() =>
+      _DownloadableResourcesPageState();
+}
+
+class _DownloadableResourcesPageState extends State<DownloadableResourcesPage> {
+  var _isDownloading = false;
+  var _isComplete = false;
+  var _isCancelled = false;
+  var _progress = 0.0;
+  
+  // To manage cancellation of download
+  CancelableOperation<List<ResponseInfo>>? _cancelableDownload;
+
+  @override
+  void initState() {
+    super.initState();
+    _checkIfResourcesExist();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.sampleTitle),
+        leading: IconButton(
+          icon: const Icon(Icons.arrow_back),
+          onPressed: () {
+            if (_isDownloading) {
+              unawaited(_cancelDownload());
+            }
+            context.pop();
+          },
+        ),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(40),
+        child: Center(
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              Text(
+                'This sample requires downloadable resources',
+                style: Theme.of(context).textTheme.titleMedium,
+                textAlign: TextAlign.center,
+              ),
+              const SizedBox(height: 20),
+              // Download progress indicator
+              Visibility(
+                visible: _isDownloading,
+                child: Column(
+                  children: [
+                    SizedBox(
+                      width: 60,
+                      height: 60,
+                      child: CircularProgressIndicator(
+                        value: _progress,
+                        backgroundColor: Colors.grey[300],
+                      ),
+                    ),
+                    const SizedBox(height: 20),
+                    Text(
+                      '${(_progress * 100).toInt()}% download complete',
+                      style: Theme.of(context).textTheme.bodyMedium,
+                      textAlign: TextAlign.center,
+                    ),
+                  ],
+                ),
+              ),
+              Visibility(
+                visible: _isComplete,
+                child: Text(
+                      'The resources have been downloaded.',
+                      style: Theme.of(context).textTheme.bodySmall,
+                      textAlign: TextAlign.center,
+                    ),
+              ),
+              // Button [Download|Cancel|Open]
+              const SizedBox(height: 20),
+              ElevatedButton(
+                onPressed: _isDownloading
+                    ? _cancelDownload
+                    : _isComplete
+                    ? _openSample
+                    : _startDownload,
+                child: Text(
+                  _isDownloading
+                      ? 'Cancel'
+                      : _isComplete
+                      ? 'Open'
```

Comment:
I guess we don't need this anymore? On Android I saw the 'open' button display briefly before the sample then loaded.